IdentityCore/src/validation/MSIDAuthority.m

```diff
@@ -137,4 +196,159 @@ + (NSURL *)cacheUrlForAuthority:(NSURL *)authority
     return [NSURL URLWithString:[NSString stringWithFormat:@"https://%@/%@", [authority msidHostWithPortIfNecessary], tenantId]];
 }
 
++ (void)resolveAuthority:(NSURL *)authority
+       userPrincipalName:(NSString *)upn
+                validate:(BOOL)validate
+                 context:(id<MSIDRequestContext>)context
+         completionBlock:(MSIDAuthorityInfoBlock)completionBlock
+{
+    NSParameterAssert(completionBlock);
+    
+    NSError *error;
+    if (![self isAuthorityFormatValid:authority context:nil error:&error])
+    {
+        completionBlock(nil, NO, error);
+        return;
+    }
+    
+    if (error)
+    {
+        completionBlock(nil, NO, error);
+        return;
+    }
+    
+    id <MSIDAuthorityResolving> resolver;
+    // ADFS.
+    if ([MSIDAuthority isADFSInstanceURL:authority])
+    {
+        resolver = [MSIDAdfsAuthorityResolver new];
+    }
+    // B2C.
+    else if ([MSIDAuthority isB2CInstanceURL:authority])
+    {
+        resolver = [MSIDB2CAuthorityResolver new];
+    }
+    // AAD.
+    else
+    {
+        resolver = [MSIDAadAuthorityResolver new];
+    }
+    
+    [resolver resolveAuthority:authority
+             userPrincipalName:upn
+                      validate:validate
+                       context:context
+               completionBlock:completionBlock];
+}
+
++ (void)loadOpenIdConfigurationInfo:(NSURL *)openIdConfigurationEndpoint
+                            context:(id<MSIDRequestContext>)context
+                    completionBlock:(MSIDOpenIdConfigurationInfoBlock)completionBlock
+{
+    NSParameterAssert(completionBlock);
```

Comment:
is openIdConfigurationEndpoint always non-nil?

Reply:
added verification for nil openIdConfigurationUrl